kernel/kernel-api/src/main/java/io/delta/kernel/internal/ActionCommitLog.java

```diff
@@ -0,0 +1,272 @@
+    if (commitFiles.isEmpty()) {
+      throw noCommitFilesFoundForVersionRange(tablePath.toString(), startVersion, endVersion);
+    }
+
+    // Verify commit files found
+    // (check that they are continuous and start with startVersion and end with endVersion)
+    verifyDeltaVersions(commitFiles, startVersion, endVersion, tablePath);
+
+    return commitFiles;
+  }
+
+  /**
+   * Read the given commitFiles and return the contents as an iterator of batches. Also adds two
+   * columns "version" and "timestamp" that store the commit version and timestamp for the commit
+   * file that the batch was read from.
+   *
+   * @param commitFiles list of delta commit files to read
+   * @param readSchema JSON schema to read
+   * @return an iterator over the contents of the files in the same order as the provided files
+   */
+  public static CloseableIterator<ColumnarBatch> readCommitFiles(
+      Engine engine, List<FileStatus> commitFiles, StructType readSchema) {
+
+    return new ActionsIterator(engine, commitFiles, readSchema, Optional.empty())
+        .map(
+            actionWrapper -> {
+              long timestamp =
+                  actionWrapper
+                      .getTimestamp()
+                      .orElseThrow(
+                          () ->
+                              new RuntimeException("Commit files should always have a timestamp"));
+              ExpressionEvaluator commitVersionGenerator =
+                  wrapEngineException(
+                      () ->
+                          engine
+                              .getExpressionHandler()
+                              .getEvaluator(
+                                  readSchema,
+                                  Literal.ofLong(actionWrapper.getVersion()),
+                                  LongType.LONG),
+                      "Get the expression evaluator for the commit version");
+              ExpressionEvaluator commitTimestampGenerator =
+                  wrapEngineException(
+                      () ->
+                          engine
+                              .getExpressionHandler()
+                              .getEvaluator(readSchema, Literal.ofLong(timestamp), LongType.LONG),
+                      "Get the expression evaluator for the commit timestamp");
+              ColumnVector commitVersionVector =
+                  wrapEngineException(
+                      () -> commitVersionGenerator.eval(actionWrapper.getColumnarBatch()),
+                      "Evaluating the commit version expression");
+              ColumnVector commitTimestampVector =
+                  wrapEngineException(
+                      () -> commitTimestampGenerator.eval(actionWrapper.getColumnarBatch()),
+                      "Evaluating the commit timestamp expression");
+
+              return actionWrapper
+                  .getColumnarBatch()
+                  .withNewColumn(0, COMMIT_VERSION_STRUCT_FIELD, commitVersionVector)
```

Comment:
this adds it at the front, right? and doesn't just overwrite what was already there?